integration/lifecycle/capabilities_endorse_test.go

```diff
@@ -0,0 +1,261 @@
+			ChannelConfigPolicy: "/Channel/Application/Endorsement",
+			Sequence:            "1",
+			InitRequired:        true,
+			Label:               "my_simple_chaincode",
+		}
+
+		By("setting up the channel with v3_0 capabilities and without the ed25519 peer")
+		network.CreateAndJoinChannels(orderer)
+		network.UpdateChannelAnchors(orderer, "testchannel")
+		network.VerifyMembership(network.PeersWithChannel("testchannel"), "testchannel")
+		nwo.EnableCapabilities(network, "testchannel", "Application", "V2_0", orderer, network.Peer("Org1", "peer0"), network.Peer("Org2", "peer0"))
+		nwo.EnableChannelCapabilities(network, "testchannel", "V3_0", false, orderer, []*nwo.Orderer{orderer},
+			network.Peer("Org1", "peer0"),
+			network.Peer("Org2", "peer0"),
+		)
+
+		By("joining the ed25519 peer to the channel")
+		ed25519Peer := network.Peer("Org1", "peer1")
+		ed25519Peer.Channels = []*nwo.PeerChannel{
+			{Name: "testchannel", Anchor: false},
+		}
+		network.JoinChannel("testchannel", orderer, ed25519Peer)
+		testPeers = append(testPeers, ed25519Peer)
+
+		By("deploying the chaincode")
+		nwo.PackageChaincodeBinary(chaincode)
+		chaincode.SetPackageIDFromPackageFile()
+
+		nwo.InstallChaincode(network, chaincode, testPeers...)
+
+		By("verifying the installed chaincode package matches the one that was submitted")
+		sess, err := network.PeerAdminSession(testPeers[0], commands.ChaincodeGetInstalledPackage{
+			PackageID:       chaincode.PackageID,
+			OutputDirectory: testDir,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+		fileBytes, err := ioutil.ReadFile(chaincode.PackageFile)
+		Expect(err).NotTo(HaveOccurred())
+		fileBytesFromPeer, err := ioutil.ReadFile(filepath.Join(network.RootDir, chaincode.PackageID+".tar.gz"))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(fileBytesFromPeer).To(Equal(fileBytes))
+
+		nwo.ApproveChaincodeForMyOrg(network, "testchannel", orderer, chaincode, testPeers...)
+
+		nwo.CheckCommitReadinessUntilReady(network, "testchannel", chaincode, network.PeerOrgs(), testPeers...)
+		nwo.CommitChaincode(network, "testchannel", orderer, chaincode, testPeers[0], testPeers...)
+		nwo.InitChaincode(network, "testchannel", orderer, chaincode, testPeers...)
+
+		By("invoking the chaincode with the ed25519 endorser")
+		endorsers := []*nwo.Peer{
+			network.Peer("Org1", "peer1"),
+			network.Peer("Org2", "peer0"),
+		}
+		RunQueryInvokeQuery(network, orderer, "My_1st-Chaincode", 100, endorsers...)
+
+		By("downgrading the channel capabilities back to v2_0")
+		nwo.EnableChannelCapabilities(network, "testchannel", "V2_0", false, orderer, []*nwo.Orderer{orderer},
+			network.Peer("Org1", "peer0"),
+			network.Peer("Org2", "peer0"),
+		)
+
+		By("invoking the chaincode again, but expecting a failure")
+		RunInvokeAndExpectFailure(network, orderer, "My_1st-Chaincode", "(ENDORSEMENT_POLICY_FAILURE)", endorsers...)
```

Comment:
can we be more specific about the failure? Is there any signature specific failure in the log of the node that we can capture here?

Reply:
The whole return message is: "Error: transaction invalidated with status (ENDORSEMENT_POLICY_FAILURE) - proposal response: <nil>". The test is meant to demonstrate that an ed25519 cannot endorse transactions in a channel with V2_0 capabilities.